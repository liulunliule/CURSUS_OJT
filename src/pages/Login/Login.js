// import React from "react";
// import renderer from "react-test-renderer";
// import "@testing-library/jest-dom/extend-expect";
// import { Provider } from "react-redux";
// import { MemoryRouter } from "react-router-dom"; // Import MemoryRouter
// import { store } from "../../redux/store";
// import Login from ".";
// import { fireEvent, render, screen } from "@testing-library/react";
// import axios from "axios";
// import { toast } from "react-toastify";
// import { fetchUserLoginSuccess } from "../../redux/features/userSlice";

// test("Login renders correctly and matches snapshot", () => {
//     const { asFragment } = render(<Login />);
//     expect(asFragment()).toMatchSnapshot();
// });

// test("renders Login Page correctly", () => {
//     render(
//         <Provider store={store}>
//             <MemoryRouter>
//                 <Login />
//             </MemoryRouter>
//         </Provider>
//     );

//     expect(screen.getByPlaceholderText("Email Address")).toBeInTheDocument();
//     expect(screen.getByPlaceholderText("Password")).toBeInTheDocument();
//     expect(screen.getByText("Sign in")).toBeInTheDocument();
//     expect(screen.getByText("Forgot Password")).toBeInTheDocument();
//     expect(screen.getByText("Don't have an account?")).toBeInTheDocument();
//     expect(screen.getByText("Sign Up")).toBeInTheDocument();
// });

// test("handles user input and login process", async () => {
//     const mockUser = { email: "huyngocnghech@gmail.com", password: "123" };
//     axios.get.mockResolvedValue({ data: [mockUser] });

//     render(
//         <Provider store={store}>
//             <MemoryRouter>
//                 <Login />
//             </MemoryRouter>
//         </Provider>
//     );

//     fireEvent.change(screen.getByPlaceholderText("Email Address"), {
//         target: { value: "huyngocnghech@gmail.com" },
//     });
//     fireEvent.change(screen.getByPlaceholderText("Password"), {
//         target: { value: "123" },
//     });

//     fireEvent.click(screen.getByText("Sign in"));

//     expect(await screen.findByText("Login succeed")).toBeInTheDocument();

//     expect(fetchUserLoginSuccess).toHaveBeenCalledWith(mockUser);
// });
