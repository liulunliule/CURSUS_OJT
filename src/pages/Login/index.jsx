import React, { useState } from "react";
import "./index.scss";
import { logo } from "../../assets";
import { Link, useNavigate } from "react-router-dom";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
    faFacebookF,
    faGoogle,
    faTwitter,
} from "@fortawesome/free-brands-svg-icons";
import { faEnvelope } from "@fortawesome/free-regular-svg-icons";
import { faKey } from "@fortawesome/free-solid-svg-icons";
import { logo_Small } from "../../assets";
import { postLogin } from "../../services/apiService";
import { message } from "antd";
import { useDispatch } from "react-redux";
import { fetchUserLoginSuccess } from "../../redux/features/userSlice";
import { LoadingOutlined } from "@ant-design/icons";
import { toast } from "react-toastify";
import axios from "axios";

function Login() {
    const [email, setEmail] = useState("");
    const [password, setPassword] = useState("");
    const dispatch = useDispatch();
    const [isLoading, setIsLoading] = useState(false);
    const navigate = useNavigate();
    const validateEmail = (email) => {
        return String(email)
            .toLowerCase()
            .match(
                /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|.(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
            );
    };
    const handleLogin = async (email, password) => {
        //    Validate
        const isValidEmail = validateEmail(email);
        if (!isValidEmail) {
            toast.error("Invalid email! Please try again");
            return;
        } else if (!password) {
            toast.error("Invalid password");
            return;
        }
        setIsLoading(true);

        // submit login
        try {
            const response = await axios.get(
                "https://6696231a0312447373c1386e.mockapi.io/user"
            );
            console.log(response.data);
            const user = response.data.find(
                (user) => user.email === email && user.password === password
            );
            if (user) {
                console.log("Check ", user);
                setIsLoading(false);
                toast.success("Login succeed");
                dispatch(fetchUserLoginSuccess(user));
                navigate("/");
            }

            if (user.password !== password) {
                toast.error("Incorrect Password. Please try again!!");
            }
        } catch (error) {
            toast.error(error.message);
            setIsLoading(false);
        }
    };

    return (
        <div>
            <div className="login">
                <Link to="/">
                    <img src={logo} alt="" className="login__logo" />
                </Link>
                <div className="login__main main">
                    <h2 className="login__title">Welcome Back</h2>
                    <p className="login__desc">
                        Log In to Your Cursus Account!
                    </p>
                    <div className="login__cta">
                        <button className="login__button">
                            <FontAwesomeIcon
                                icon={faFacebookF}
                                style={{
                                    color: "#ffffff",
                                    marginRight: "10px",
                                }}
                            />
                            Continue with Facebook
                        </button>
                        <button className="login__button">
                            <FontAwesomeIcon
                                icon={faTwitter}
                                style={{
                                    color: "#ffffff",
                                    marginRight: "10px",
                                }}
                            />
                            Continue with Twitter
                        </button>
                        <button className="login__button">
                            <FontAwesomeIcon
                                icon={faGoogle}
                                style={{
                                    color: "#ffffff",
                                    marginRight: "10px",
                                }}
                            />
                            Continue with Google
                        </button>
                        <div className="login__form-group">
                            <div className="login__form">
                                <FontAwesomeIcon
                                    icon={faEnvelope}
                                    style={{ color: "#909090" }}
                                    className="faEnvelope"
                                />
                                <input
                                    type="text"
                                    name=""
                                    id=""
                                    placeholder="Email Address"
                                    className="login__form-input"
                                    value={email}
                                    onChange={(event) =>
                                        setEmail(event.target.value)
                                    }
                                />
                            </div>

                            <div className="login__form">
                                <FontAwesomeIcon
                                    icon={faKey}
                                    style={{ color: "#909090" }}
                                    className="faKey"
                                />
                                <input
                                    type="password"
                                    name=""
                                    id=""
                                    placeholder="Password"
                                    className="login__form-input"
                                    value={password}
                                    onChange={(event) =>
                                        setPassword(event.target.value)
                                    }
                                />
                            </div>
                            <div className="login__remember">
                                <input
                                    type="checkbox"
                                    name=""
                                    id="remember-me"
                                />
                                <label htmlFor="remember-me">Remember me</label>
                            </div>
                            <button
                                className="login__button login__signin"
                                onClick={() => handleLogin(email, password)}
                                disabled={isLoading}
                            >
                                {isLoading === true && (
                                    <LoadingOutlined
                                        style={{ marginRight: "10px" }}
                                    />
                                )}
                                Sign in
                            </button>
                        </div>
                    </div>

                    <div className="login__forgotPass">
                        Or{" "}
                        <Link to="/forgot_password" className="aaaa">
                            {" "}
                            Forgot Password
                        </Link>
                    </div>
                    <div className="login__line"></div>
                    <div className="login__signup">
                        Don't have an account?
                        <Link className="login__signup-link" to="/signup">
                            Sign Up
                        </Link>
                    </div>
                </div>
                <div className="login__footer">
                    <img src={logo_Small} alt="" />
                    <p className="login__footer-content">
                        Â© 2024 <b>Cursus</b>. All Rights Reserved.
                    </p>
                </div>
            </div>
        </div>
    );
}

export default Login;
