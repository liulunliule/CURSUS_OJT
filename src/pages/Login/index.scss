.login {
    display: flex;
    position: relative;
    flex-direction: column;
    align-items: center;
    padding: 50px 0;
    background-color: #f7f7f7;
    z-index: 1;
    &::before {
        content: "";
        background: url(../../assets/img/backgroundLDP.svg) no-repeat center;
        inset: 0;
        position: absolute;
        background-size: cover;
        opacity: 0.04;
        z-index: -1;
    }

    &__logo {
        width: 135px;
    }

    &__title {
        color: #333;
        font-size: 24px;
        font-weight: 700;
        line-height: 30.8571px;
    }

    &__desc {
        margin-top: 17px;
        color: #686f7a;
        font-size: 14px;
        line-height: 26px;
    }
    &__cta {
        display: flex;
        margin-top: 47px;
        flex-direction: column;
        gap: 15px;
    }
    &__button {
        padding: 10px 148px;
        border: none;
        outline: none;
        font-size: 14px;
        color: #fff;
        background-color: #3b5998;
        border-radius: 3px;
        line-height: 16.1px;

        &:hover {
            cursor: pointer;
        }
    }
    &__button:nth-child(2) {
        background-color: #1da1f2;
    }
    &__button:nth-child(3) {
        background-color: #34a853;
    }
    &__signin {
        margin-top: 31px;
        background-color: #ed2a26;
        width: 479px;

        &:hover {
            background-color: #333;
        }
    }
    &__form {
        display: flex;
        // align-items: center;
        justify-content: space-between;

        width: 479px;
        height: 40px;
        border: 1px solid #e5e5e5;
        border-radius: 3px;
        padding: 9.5px 20px;
        margin-bottom: 15px;
        transition: all 0.2s;

        &-input {
            width: 100%;
            height: 100%;
            border: none;
            outline: none;
            padding: 0 20px;
            font-size: 14px;

            &::placeholder {
                color: #c9c9c9;
            }
        }
        &:focus-within {
            border-color: #333;

            .faEnvelope,
            .faKey {
                color: #333;
            }

            .login__form-input::placeholder {
                color: #999494;
            }
        }
    }
    &__remember {
        margin-top: 30px;
        display: flex;
        align-items: center;

        & label {
            position: relative;
            display: flex;
            align-items: center;
            user-select: none;
        }

        & label::before,
        & label::after {
            content: "";
            display: inline-block;
            margin-right: 7px;
        }

        & label::before {
            width: 17px;
            height: 17px;
            border-radius: 50%;
            border: 1px solid #e5e5e5;
        }
        & [type="checkbox"] {
            display: none;

            &:checked + label::before {
                border-color: #ed2a26;
            }
            &:checked + label::after {
                position: absolute;
                top: 8px;
                left: 4px;
                width: 8px;
                height: 5px;
                border-left: 3px solid #ed2a26;
                border-bottom: 3px solid #ed2a26;
                rotate: -45deg;
            }
        }
    }
    &__signup,
    &__forgotPass {
        margin-top: 51px;
        color: #686f7a;
        font-size: 14px;

        .aaaa{
            text-decoration: none;
        }

        & a {
            color: red;
            text-decoration: none;

            &:hover {
                color: #333;
                text-decoration: underline;
            }
        }
    }
    &__line {
        margin-top: 36px;
        width: 479px;
        height: 1px;
        background-color: #e5e5e5;
    }
    &__signup {
        margin-top: 30px;
        & a {
            margin-left: 3px;
        }
    }
    &__footer {
        display: flex;
        align-items: center;
        margin-top: 57px;

        &-content {
            margin: 0;
        }
    }
}
