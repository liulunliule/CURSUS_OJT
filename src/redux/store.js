import { combineReducers, configureStore } from "@reduxjs/toolkit";
import savedCourseReducer from "./features/savedCourseSlice";
import blogReducer from './features/ourBlogSlice';
import officesReducer from './features/officesSlice';
import pressReducer from './features/pressSlice';
import payoutReducer from './features/payoutSlice';
import creditsReducer from "./features/creditsSlice";
import certificateReducer from "./features/myCertificateSlice";
import myProfileReducer from "./features/myProfileSlice";
import myHeaderReducer from "./features/myHeaderSlice";
import mySearchReducer from "./features/mySearchSlice";
import myStudioReducer from "./features/myStudioSlice";
import allInstructorReducer from "./features/allInstructorSlice";
import reviewFeedbackReducer from "./features/reviewfeedbackSlice";
import feedbackReportReducer from "./features/feedbackvsReportSlice";
import notificationsSettingReducer from "./features/NotificationsSettingSlice";
import privacySettingReducer from "./features/PrivacySettingSlice";
import userSlice from "./features/userSlice";
import certificationTestPageReducer from "./features/certificationTestPageSlice";
import certificationTestResultReducer from "./features/certificationTestResultSlice";
import analyticsReducer from "./features/analyticsSlice";
import storage from "redux-persist/lib/storage";
import { persistReducer, persistStore } from "redux-persist";
import courseSlice from "./features/courseSlice";
import checkoutReducer from './features/checkoutSlice';
import earningReducer from './features/earningSlice';
import statementReducer from './features/statementSlice';
import shoppingCartReducer from './features/shoppingCartSlice';

const persistConfig = {
    key: "root",
    storage,
    whitelist: [
        "user",
        "savedCourse",
        "certificate",
        "course",
        "myProfile",
        "myHeader",
        "mySearch",
        "myStudio",
        "allInstructor",
        "reviewFeedback",
        "feedbackReport",
        "notificationsSetting",
        "privacySetting",
    ],
};

const rootReducer = combineReducers({
    user: userSlice,
    savedCourse: savedCourseReducer,
    certificate: certificateReducer,
    certificationTestPage: certificationTestPageReducer,
    certificationTestResult: certificationTestResultReducer,
    analytics: analyticsReducer,
    course: courseSlice,
    myProfile: myProfileReducer,
    myHeader: myHeaderReducer,
    mySearch: mySearchReducer,
    myStudio: myStudioReducer,
    allInstructor: allInstructorReducer,
    reviewFeedback: reviewFeedbackReducer,
    feedbackReport: feedbackReportReducer,
    checkout: checkoutReducer,
    earnings: earningReducer,
    statement: statementReducer,
    credits: creditsReducer,
    payout: payoutReducer,
    blogs: blogReducer,
    offices: officesReducer,
    press: pressReducer,
    notificationsSetting: notificationsSettingReducer,
    privacySetting: privacySettingReducer,
    shoppingCart: shoppingCartReducer,
});

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
    reducer: persistedReducer,
});

export const persistor = persistStore(store);
