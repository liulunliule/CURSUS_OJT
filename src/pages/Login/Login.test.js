import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom'; // Import MemoryRouter
import { store } from '../../redux/store';
import Login from '.';

test('test Login Page', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <Login/>
      </MemoryRouter>
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
