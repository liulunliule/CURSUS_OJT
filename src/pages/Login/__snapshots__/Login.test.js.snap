// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Login Page 1`] = `
<div>
  <div
    className="login"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt=""
        className="login__logo"
        src="logo.svg"
      />
    </a>
    <div
      className="login__main main"
    >
      <h2
        className="login__title"
      >
        Welcome Back
      </h2>
      <p
        className="login__desc"
      >
        Log In to Your Cursus Account!
      </p>
      <div
        className="login__cta"
      >
        <button
          className="login__button"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-facebook-f "
            data-icon="facebook-f"
            data-prefix="fab"
            focusable="false"
            role="img"
            style={
              Object {
                "color": "#ffffff",
                "marginRight": "10px",
              }
            }
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M80 299.3V512H196V299.3h86.5l18-97.8H196V166.9c0-51.7 20.3-71.5 72.7-71.5c16.3 0 29.4 .4 37 1.2V7.9C291.4 4 256.4 0 236.2 0C129.3 0 80 50.5 80 159.4v42.1H14v97.8H80z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          Continue with Facebook
        </button>
        <button
          className="login__button"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-twitter "
            data-icon="twitter"
            data-prefix="fab"
            focusable="false"
            role="img"
            style={
              Object {
                "color": "#ffffff",
                "marginRight": "10px",
              }
            }
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          Continue with Twitter
        </button>
        <button
          className="login__button"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-google "
            data-icon="google"
            data-prefix="fab"
            focusable="false"
            role="img"
            style={
              Object {
                "color": "#ffffff",
                "marginRight": "10px",
              }
            }
            viewBox="0 0 488 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M488 261.8C488 403.3 391.1 504 248 504 110.8 504 0 393.2 0 256S110.8 8 248 8c66.8 0 123 24.5 166.3 64.9l-67.5 64.9C258.5 52.6 94.3 116.6 94.3 256c0 86.5 69.1 156.6 153.7 156.6 98.2 0 135-70.4 140.8-106.9H248v-85.3h236.1c2.3 12.7 3.9 24.9 3.9 41.4z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          Continue with Google
        </button>
        <div
          className="login__form-group"
        >
          <div
            className="login__form"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-envelope faEnvelope"
              data-icon="envelope"
              data-prefix="far"
              focusable="false"
              role="img"
              style={
                Object {
                  "color": "#909090",
                }
              }
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M64 112c-8.8 0-16 7.2-16 16l0 22.1L220.5 291.7c20.7 17 50.4 17 71.1 0L464 150.1l0-22.1c0-8.8-7.2-16-16-16L64 112zM48 212.2L48 384c0 8.8 7.2 16 16 16l384 0c8.8 0 16-7.2 16-16l0-171.8L322 328.8c-38.4 31.5-93.7 31.5-132 0L48 212.2zM0 128C0 92.7 28.7 64 64 64l384 0c35.3 0 64 28.7 64 64l0 256c0 35.3-28.7 64-64 64L64 448c-35.3 0-64-28.7-64-64L0 128z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            <input
              className="login__form-input"
              id=""
              name=""
              onChange={[Function]}
              placeholder="Email Address"
              type="text"
              value=""
            />
          </div>
          <div
            className="login__form"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-key faKey"
              data-icon="key"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={
                Object {
                  "color": "#909090",
                }
              }
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M336 352c97.2 0 176-78.8 176-176S433.2 0 336 0S160 78.8 160 176c0 18.7 2.9 36.8 8.3 53.7L7 391c-4.5 4.5-7 10.6-7 17l0 80c0 13.3 10.7 24 24 24l80 0c13.3 0 24-10.7 24-24l0-40 40 0c13.3 0 24-10.7 24-24l0-40 40 0c6.4 0 12.5-2.5 17-7l33.3-33.3c16.9 5.4 35 8.3 53.7 8.3zM376 96a40 40 0 1 1 0 80 40 40 0 1 1 0-80z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            <input
              className="login__form-input"
              id=""
              name=""
              onChange={[Function]}
              placeholder="Password"
              type="text"
              value=""
            />
          </div>
          <div
            className="login__remember"
          >
            <input
              id="remember-me"
              name=""
              type="checkbox"
            />
            <label
              htmlFor="remember-me"
            >
              Remember me
            </label>
          </div>
          <button
            className="login__button login__signin"
            disabled={false}
            onClick={[Function]}
          >
            Sign in
          </button>
        </div>
      </div>
      <div
        className="login__forgotPass"
      >
        Or
         
        <a
          className="aaaa"
          href="/forgot_password"
          onClick={[Function]}
        >
           
          Forgot Password
        </a>
      </div>
      <div
        className="login__line"
      />
      <div
        className="login__signup"
      >
        Don't have an account?
        <a
          className="login__signup-link"
          href="/signup"
          onClick={[Function]}
        >
          Sign Up
        </a>
      </div>
    </div>
    <div
      className="login__footer"
    >
      <img
        alt=""
        src="small_logo.png"
      />
      <p
        className="login__footer-content"
      >
        Â© 2024 
        <b>
          Cursus
        </b>
        . All Rights Reserved.
      </p>
    </div>
  </div>
</div>
`;
